adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -503,19 +605,53 @@ case class ReferenceRegion(
   }
 
   /**
+   * Determines if two regions are colocated on the same strand.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same strand, false otherwise
+   */
+  def sameStrand(other: ReferenceRegion): Boolean = {
+    strand == other.strand
+  }
+
+  /**
+   * Determines if two regions are colocated on the same reference name.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same reference name, false otherwise.
+   */
+  def sameReferenceName(other: ReferenceRegion): Boolean = {
+    referenceName == other.referenceName
+  }
+
+  /**
    * @return The length of this region in bases.
    */
   def length(): Long = {
     end - start
   }
 
   override def hashCode: Int = {
-    var result = 37
-    result = 41 * result + (if (referenceName != null) referenceName.hashCode else 0)
-    result = 41 * result + start.hashCode
-    result = 41 * result + end.hashCode
-    result = 41 * result + (if (strand != null) strand.ordinal() else 0)
-    result
+    // add 37 here because this is the start of the hashCode
+    val nameHashCode = 37 + {
+      if (referenceName != null) {
```

Comment:
Ah, I see what's going on. The [createUnmappedRead](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/MarkDuplicatesSuite.scala#L36-L40) function in the test code is creating reads without sequences. Can you update that function to set the sequence on the read (doesn't matter what the sequence is, just has to be present)? We should probably add a `require(read.getSequence != null` [right before here](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/ReferencePositionPair.scala#L53).